src/Oscoin/Deployment/Internal/CloudInit.hs

```diff
@@ -0,0 +1,241 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module Oscoin.Deployment.Internal.CloudInit
+    ( CloudConfig
+    , mkCloudConfig
+    , renderCloudConfig
+
+    , User
+    , mkUser
+    , rootUser
+    , renderUidOrName
+
+    , Group
+    , mkGroup
+    , rootGroup
+    , renderGidOrName
+
+    , Owner
+    , mkOwner
+    , rootOwner
+
+    , WriteFiles
+    , mkWriteFiles
+
+    , FileContents
+    , plainTextFileContents
+    , base64FileContents
+
+    , FileMode
+    , fromPosixFileMode
+    )
+where
+
+import           Oscoin.Prelude hiding (group)
+
+import           Data.Aeson (ToJSON(..), (.=))
+import qualified Data.Aeson.Types as Aeson
+import           Data.ByteString.BaseN (AtBase, encodedText)
+import           Data.Generics.Product
+import qualified Data.Set as Set
+import           Data.Vector (Vector)
+import qualified Data.Yaml as Yaml
+import           Lens.Micro (set)
+import           Lens.Micro.Extras (view)
+import           Numeric (showIntAtBase)
+import qualified System.Posix as Posix
+
+data CloudConfig = CloudConfig
+    { users       :: Vector User
+    , groups      :: Vector Group
+    , write_files :: Vector WriteFiles
+    , runcmd      :: Vector Text
+    , bootcmd     :: Vector Text
```

Comment:
Genuine question -- what's the advantage of using vectors here?

Reply:
[Fewer allocations](https://hackage.haskell.org/package/aeson-1.4.3.0/docs/Data-Aeson.html#t:Array)

Reply:
I see -- didn't think these were a concern for a deployment tool, good to know 😅

Reply:
Conversely: what's the issue with vectors? They're almost always a better default than lists. One could argue that `users` and `groups` should be `Set`, but that would require unsound `Eq` instances.